File: go/p2.go
package main

import (
  "fmt"
)

func evenFibs(max int) int {
  var first, second int = 1, 2
  var sum int = 0

  for second < max {
    if second % 2 == 0 {
      sum += second
      fmt.Printf("%d\n", second)
    }

    first, second = second, first+second
  }

  return sum
}

func main() {
  fmt.Printf("%d\n", evenFibs(4000000))
}

File: js/p2.js
function sumEvenFibs(max) {
  var first = 1, second = 2;
  var sum = 0;

  while (second < max) {
    if (second % 2 === 0) {
      sum += second;
    }

    var tmp = second;
    second += first;
    first = tmp;
  }

  return sum
}

console.log( sumEvenFibs(4000000) );
